File: frontend/src/features/projects/types/project.ts
export enum ProjectStatus {
  Planning = 0,
  Active = 1,
  Completed = 2,
  OnHold = 3,
  Cancelled = 4
}

export interface Project {
  id: number;
  code: string;
  name: string;
  departmentId?: number;
  managerId?: number;
  budget: number;
  startDate?: string;
  endDate?: string;
  status: ProjectStatus;
  createdAt: string;
}

export interface ProjectFinancials {
  projectId?: number;
  projectCode: string;
  budget: number;
  totalRevenue: number;
  totalCost: number;
  profit: number;
  budgetRemaining: number;
  details: {
    salesCount: number;
    purchaseCount: number;
    voucherCount: number;
  };
}

File: backend/src/ERP.Modules.HR/Models/Project.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ERP.Modules.HR.Models;

public enum ProjectStatus
{
    Planning = 0,
    Active = 1,
    Completed = 2,
    OnHold = 3,
    Cancelled = 4
}

public class Project
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    [MaxLength(50)]
    public string Code { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public int? DepartmentId { get; set; }
    [ForeignKey("DepartmentId")]
    public Department? Department { get; set; }
    
    public int? ManagerId { get; set; }
    [ForeignKey("ManagerId")]
    public Employee? Manager { get; set; }
    
    [Column(TypeName = "decimal(18,2)")]
    public decimal Budget { get; set; }
    
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    
    public ProjectStatus Status { get; set; } = ProjectStatus.Planning;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
